linux/guest_customization/check_gosc_support_status.yml

```diff
@@ -89,25 +90,44 @@
               https://bugs.launchpad.net/cloud-init/+bug/1944946.
               So skip cloud-init GOSC testing for {{ vm_guest_os_distribution }}.
       when:
-        - enable_cloudinit_gosc | bool
+        - enable_cloudinit_gosc
         - guest_os_ansible_distribution == "Debian"
         - guest_os_ansible_distribution_major_ver | int == 12
 
     - name: "Check Perl GOSC support status for {{ vm_guest_os_distribution }}"
       when:
-        - not enable_cloudinit_gosc | bool
+        - not enable_cloudinit_gosc
         - guest_os_ansible_distribution not in gos_not_support_gosc
       block:
-        - name: "Get GOSC support matrix"
+        - name: "Set facts of Perl GOSC support matrix"
           ansible.builtin.set_fact:
             gosc_support_matrix: "{{ lookup('file', 'gosc_support_matrix.yml') | from_yaml }}"
+            guest_gosc_suport_matrix: {}
+
+        # RHEL family OS 9.0 or higher version aligns with GOSC support status for same version of RHEL
+        - name: "Set fact of Perl GOSC matrix for {{ vm_guest_os_distribution }}"
+          ansible.builtin.set_fact:
+            guest_gosc_suport_matrix: "{{ gosc_support_matrix[guest_os_family] | combine(gosc_support_matrix[guest_os_ansible_distribution] | default({})) }}"
+          when:
+            - guest_os_family == "RedHat"
+            - guest_os_ansible_distribution != 'MIRACLE'
+            - guest_os_ansible_distribution_major_ver | int >= 9
+
+        - name: "Set fact of Perl GOSC matrix for {{ vm_guest_os_distribution }}"
+          ansible.builtin.set_fact:
+            guest_gosc_suport_matrix: "{{ gosc_support_matrix[guest_os_ansible_distribution] }}"
+          when:
+            - (guest_os_family != "RedHat" or
+               guest_os_ansible_distribution == 'MIRACLE' or
+               guest_os_ansible_distribution_major_ver | int < 9)
+            - gosc_support_matrix[guest_os_ansible_distribution] is defined
```

Comment:
Can this line also be set as the '| default({})' as above?

Reply:
Line 105 has already initialized the value of `guest_gosc_suport_matrix` to `{}`.  Here will only change it when `gosc_support_matrix[guest_os_ansible_distribution]` is defined in 'gosc_support_matrix.yml' for Miracle Linux and Linux other than RedHat family.

The reason to use `default({})` in above set_fact is mainly for OS like CentOS and Oracle Linux, which have completely same support matrix with RHEL but don't have their own definitions. Using `gosc_support_matrix[guest_os_ansible_distribution] | default({}` to avoid undefined error.

Reply:
Simplified setting guest_gosc_suport_matrix into 2 `set_facts` tasks instead of 3.

Reply:
Just realized that it can't be simplified as `guest_gosc_suport_matrix` cannot use `gosc_support_matrix` in the same `set_fact` task. So I reverted the change